packages/table/src/table.tsx

```diff
@@ -356,7 +356,20 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         newRowHeights = Utils.arrayOfLength(newRowHeights, numRows, defaultRowHeight);
         newRowHeights = Utils.assignSparseValues(newRowHeights, rowHeights);
 
-        const newSelectedRegions = (selectedRegions == null) ? this.state.selectedRegions : selectedRegions;
+        const numCols = newColumnWidths.length;
+
+        let newSelectedRegions = selectedRegions;
+        if (selectedRegions == null) {
```

Comment:
leave a comment explaining that this is forking logic based on controlled vs. uncontrolled